library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/obfuscator/rules/TextValueObfuscationRule.kt

```diff
@@ -0,0 +1,18 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.obfuscator.rules
+
+import android.widget.TextView
+
+/**
+ * Will be used to apply the masking strategy for a [TextView] following the rules defined
+ * in the document: [Privacy Options Rules](https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/2945942237/Privacy+Options+in+Mobile+Session+Replay)
+ */
+internal interface TextValueObfuscationRule {
+
+    fun resolveObfuscatedValue(textView: TextView): String
```

Comment:
question/ Can we assume uniform behaviour for all `TextView` instances? Two negative examples:
* what if we want to mask differently `TextView` elements displayed in calendar vs in scroll view?
* in "fine-grained masking" feature, we will likely decide to [allow overriding masking behaviour in child elements](https://docs.datadoghq.com/real_user_monitoring/session_replay/privacy_options/#override-an-html-element) from their container - how would we know it here, given that we only have the access to `TextView` instance, not its container?

Reply:
I thought of that too and I am going to introduce later a new type `OPTION_IN_PARENT_SELECTOR` which I am going to be able to resolve it from the `TextTypeResolver` by checking the parent type of the `TextView`. In Android the `TextView` has a `parent` property that I can leverage for this.

Reply:
That would mean we'll be effectively iterating the view tree not only "top (root) → bottom (leafs)" (`O(N)`) but also "bottom → up (root)" **for each view** in the hierarchy, no? We would need to recursively search `parents` in the hierarchy to find the ones that apply mask override (or traverse the entire tree up if none does it). This sounds like switching from `O(N)` to something much heavier for the main thread operation, no?